lib/filewatch/fingerprinter.rb

```diff
@@ -0,0 +1,130 @@
+require 'filewatch/boot_setup' unless defined?(FileWatch)
+
+module FileWatch
+  class Fingerprinter
+    attr_reader :data, :fnv, :fingerprint, :offset, :size
+
+    def initialize(path, offset)
+      @path = path
+      @offset = offset
+      @size = FP_BYTE_SIZE
+    end
+
+    def match_any?(keys)
+      if @fnv.nil?
+        # STDERR.puts "Fingerprinter: fnv nil for path - #{path}"
+        return nil
+      end
+      if @data.nil?
+        # STDERR.puts "Fingerprinter: data nil for path - #{path}"
+        return nil
+      end
+      keys.each do |k|
```

Comment:
k is returned on line 32.
Point taken on name abbrev.
with names of local vars - I am always wary of shadowing attr_reader methods.
`offset != @offset` or worse! `offset != self.offset`